### PR TITLE
Theme naming, ...

### LINKED ISSUE #1915: The terms of the theme should be changed
https://github.com/camptocamp/ngeo/blob/master/contribs/gmf/externs/gmf-themes.js

### PR DESCRIPTION
**Pull request by sbrunner**
fix #1915

See different commits :-)

### CODE REVIEW

File: contribs/gmf/externs/gmf-themes.js [hunk 1]
```diff
@@ -27,384 +17,457 @@ var GmfThemesResponse;
...
- * @type {Array.<GmfGroup>}
+ * @type {Array.<GmfGroup|GmfLayer>}
  */
-GmfTheme.prototype.children;
+GmfGroup.prototype.children;
 
 
 /**
- * @type {Object.<string, Array.<string|number>>}
+ * A mixed group is a group on witch one the layers comes from different sources,
```
Comment (fredj) [inline contribs/gmf/externs/gmf-themes.js]: which instead of witch

File: contribs/gmf/externs/gmf-themes.js [hunk 2]
```diff
@@ -27,384 +17,457 @@ var GmfThemesResponse;
...
+ * @struct
+ * @extends GmfLayer
  */
-GmfGroup.prototype.metadata;
+var GmfLayerWMS = function() {};
 
 
 /**
- * @type {boolean|undefined}
+ * The coma separated list of WMS layers or groups.
```
Comment (fredj) [inline contribs/gmf/externs/gmf-themes.js]: comma instead of coma

File: contribs/gmf/externs/gmf-themes.js [hunk 3]
```diff
@@ -27,384 +17,457 @@ var GmfThemesResponse;
...
- * @constructor
+ * The WMS rule used to get the icon visible in the layer tree.
+ * @type {string|undefined}
  */
-var GmfMetaData = function() {};
+GmfMetaData.prototype.legendRule;
 
 
 /**
+ * The URL to the an image used as a legend in the layer tree.
```
Comment (fredj) [inline contribs/gmf/externs/gmf-themes.js]: The URL to the ~~an~~ image used as a legend in the layer tree.

File: contribs/gmf/externs/gmf-themes.js [hunk 4]
```diff
@@ -27,384 +17,457 @@ var GmfThemesResponse;
...
+ * The icon URL visible in the layer tree.
  * @type {string|undefined}
  */
-GmfMetaData.prototype.identifierAttributeField;
+GmfMetaData.prototype.iconUrl;
 
 
-/**
- * @type {boolean|undefined}
+/**o
```
Comment (fredj) [inline contribs/gmf/externs/gmf-themes.js]: remove `o`

File: contribs/gmf/src/directives/layertree.js [hunk 5]
```diff
@@ -422,65 +430,74 @@ gmf.LayertreeController.prototype.updateWMSTimeLayerState = function(
  * string for internal WMS layers without multiple childlayers in the node.
  * @param {ngeo.LayertreeController} treeCtrl ngeo layertree controller, from
  *     the current node.
- * @return {?string} The icon legend URL or null.
+ * @return {string|undefined} The icon legend URL or null.
```
Comment (fredj) [inline contribs/gmf/src/directives/layertree.js]: change `null` to `undefined` in the description

File: contribs/gmf/src/directives/layertree.js [hunk 6]
```diff
@@ -422,65 +430,74 @@ gmf.LayertreeController.prototype.updateWMSTimeLayerState = function(
...
+  );
 };
 
 
 /**
  * Get the legend URL for the given treeCtrl.
  * @param {ngeo.LayertreeController} treeCtrl ngeo layertree controller, from
  *     the current node.
- * @return {?string} The legend URL or null.
+ * @return {string|undefined} The legend URL or null.
```
Comment (fredj) [inline contribs/gmf/src/directives/layertree.js]: change `null` to `undefined` in the description

File: src/services/layerHelper.js [hunk 7]
```diff
@@ -248,7 +248,7 @@ ngeo.LayerHelper.prototype.getLayerByName = function(layerName, layers) {
  * Get the WMTS legend URL for the given layer.
  * @param {ol.layer.Tile} layer Tile layer as returned by the
  * ngeo layerHelper service.
- * @return {?string} The legend URL or null.
+ * @return {string|undefined} The legend URL or null.
```
Comment (fredj) [inline src/services/layerHelper.js]: change `null` to `undefined` in the description

File: src/services/layerHelper.js [hunk 8]
```diff
@@ -271,13 +271,13 @@ ngeo.LayerHelper.prototype.getWMTSLegendURL = function(layer) {
  * @param {string} layerName The name of a wms layer.
  * @param {number=} opt_scale A scale.
  * @param {string=} opt_legendRule rule parameters to add to the returned URL.
- * @return {?string} The legend URL or null.
+ * @return {string|undefined} The legend URL or null.
```
Comment (fredj) [inline src/services/layerHelper.js]: change `null` to `undefined` in the description

**Comment by sbrunner** [comment]
Thanks @fredj  :-)

**Pull request merged**